war/src/main/js/add-item.js

```diff
@@ -0,0 +1,386 @@
+// Initialize all modules by requiring them. Also makes sure they get bundled (see gulpfile.js).
+var $ = require('jquery-detached').getJQuery();
+
+var getItems = function(root){
+  var d = $.Deferred();
+  $.get(root+'/categories?depth=3').done(
+      function(data){
+        d.resolve(data);
+      }
+  );
+  return d.promise();
+}; 
+
+var jRoot = $('head').attr('data-rooturl');
+
+$.when(getItems(jRoot)).done(function(data){
```

Comment:
:bug: unless I am misreading this, you are unconditionally calling `/categories` even when called from `/job/somefolder/view/All/newJob` (which should be calling `/job/somefolder/view/All/categories` so as to get a folder-specific item list).